src/main/java/carpet/script/utils/ShapesRenderer.java

```diff
@@ -233,6 +288,179 @@ public void promoteWith(RenderedShape<?> rshape)
         }
     }
 
+    public static class RenderedSprite extends RenderedShape<ShapeDispatcher.DisplayedSprite> {
+
+        private boolean isitem;
+
+        public RenderedSprite set_isitem(boolean isitem) {
+            this.isitem = isitem;
+            return this;
+        }
+
+        private BlockPos blockPos;
+        private BlockState blockState;
+        private BlockEntity BlockEntity = null;
+
+        protected RenderedSprite(Minecraft client, ShapeDispatcher.ExpiringShape shape) {
+            super(client, (ShapeDispatcher.DisplayedSprite) shape);
+        }
+
+        @Override
+        public void renderLines(PoseStack matrices, Tesselator tessellator, BufferBuilder builder, double cx, double cy,
+                double cz, float partialTick) {
+            if (shape.a == 0.0)
+                return;
+
+            Vec3 v1 = shape.relativiseRender(client.level, shape.pos, partialTick);
+            Camera camera1 = client.gameRenderer.getMainCamera();
+
+            matrices.pushPose();
+            // matrices.setIdentity();
+            if (!isitem)// blocks should use its center as the origin
+                matrices.translate(0.5, 0.5, 0.5);
+            matrices.translate(v1.x - cx, v1.y - cy, v1.z - cz);
+            if (shape.facing == null) {
+                // matrices.method_34425(new Matrix4f(camera1.getRotation()));
+                matrices.mulPose(camera1.rotation());
+                matrices.mulPose(Vector3f.YP.rotationDegrees(180));
+            } else {
+                switch (shape.facing) {
+                    case NORTH:
+                        break;
+                    case SOUTH:
+                        matrices.mulPose(Vector3f.YP.rotationDegrees(180));
+                        break;
+                    case EAST:
+                        matrices.mulPose(Vector3f.YP.rotationDegrees(270));
+                        break;
+                    case WEST:
+                        matrices.mulPose(Vector3f.YP.rotationDegrees(90));
+                        break;
+                    case UP:
+                        matrices.mulPose(Vector3f.XP.rotationDegrees(90));
+                        break;
+                    case DOWN:
+                        matrices.mulPose(Vector3f.XP.rotationDegrees(-90));
+                        break;
+                }
+            }
+            // RenderSystem.scalef(shape.size* 0.0025f, -shape.size*0.0025f,
+            // shape.size*0.0025f);
+            if (shape.tilt != 0.0f)
+                matrices.mulPose(Vector3f.ZP.rotationDegrees(shape.tilt));
+            if (shape.lean != 0.0f)
+                matrices.mulPose(Vector3f.XP.rotationDegrees(shape.lean));
+            if (shape.turn != 0.0f)
+                matrices.mulPose(Vector3f.YP.rotationDegrees(shape.turn));
+            matrices.scale(shape.scale_x, shape.scale_y, shape.scale_z);
+
+            if (!isitem)// blocks should use its center as the origin
```

Comment:
you could have mentioned that in the docs - seems important

Reply:
mentioned